core/src/style/pointStyleBuilder.cpp

```diff
@@ -304,9 +304,11 @@ bool PointStyleBuilder::addPolygon(const Polygon& _polygon, const Properties& _p
             }
         }
     } else {
-        glm::vec2 c = centroid(_polygon);
-
-        addLabel(Point{c,0}, uvsQuad, p, _rule);
+        glm::vec3 c;
+        if (!_polygon.empty()) {
+            c = centroid(_polygon.front().begin(), _polygon.front().end());
+        }
+        addLabel(c, uvsQuad, p, _rule);
```

Comment:
`glm::vec` and `glm::mat` types all have default constructors that initialize the members to zero.